src/proof/veriT/veriT_post_processor.cpp

```diff
@@ -0,0 +1,1163 @@
+    // premises: 
+    // args: ()
+    //
+    // proof rule: resolution
+    // proof term: (F)
+    // premises: (VP:(or (not (not (not F))) F)) (P:(not (not F)))
+    // args: ()
+    case PfRule::NOT_NOT_ELIM:{
+      Node vp = d_nm->mkNode(kind::OR,d_nm->mkNode(kind::NOT,children[0]),res);
+      return addVeriTStep(vp,VeriTRule::NOT_NOT,{},{},*cdp)
+             && addVeriTStep(res,VeriTRule::RESOLUTION,{children[0],vp},{},*cdp);
+    }  
+    // ======== Contradiction
+    // Children: (P1:F P2:(not F))
+    // Arguments: ()
+    // ---------------------
+    // Conclusion: false
+    //
+    // proof rule: resolution
+    // proof term: ()
+    // premises: (P1:F) (P2:(not F))
+    // args: ()
+    case PfRule::CONTRA:{
+      new_args.push_back(d_nm->mkConst<Rational>(static_cast<unsigned>(VeriTRule::RESOLUTION)));
+      new_args.push_back(Node::null()); //print this as empty	
+      cdp->addStep(res,PfRule::VERIT_RULE,children,new_args);
+      return true;	
+    }
+    // ======== And elimination
+    // Children: (P:(and F1 ... Fn))
+    // Arguments: (i)
+    // ---------------------
+    // Conclusion: (Fi)
+    //
+    // proof rule: and
+    // proof term: (Fi)
+    // premises: (P:(and F1 ... Fn))
+    // args: ()
+    case PfRule::AND_ELIM:{
+      return addVeriTStep(res,VeriTRule::AND,children,{},*cdp);
+    }
+    // ======== And introduction
+    // Children: (P1:F1 ... Pn:Fn))
+    // Arguments: ()
+    // ---------------------
+    // Conclusion: (and P1 ... Pn)
+    //
+    // First the tautology (or (and F1 ... Fn) (not F1) ... (not Fn))) is introduced. Then, the resolution rule is applied n times where Vpi+1 is obtained by applying resolution to VPi:(or (and F1 ... Fn) (not Fi) ... (not Fn))) and Fi.
+    //
+    // proof rule: and_neg
+    // proof term: (VP1:(or (and F1 ... Fn) (not F1) ... (not Fn)))
+    // premises: ()
+    // args: ()
+    //
+    // proof rule: resolution
+    // proof term: (VP2:(or (and F1 ... Fn) (not F2) ... (not Fn)))
+    // premises: (VP1:(or (and F1 ... Fn) (not F1) ... (not Fn))) P1
+    // args: ()
+    //
+    // ...
+    //
+    // proof rule: resolution
+    // proof term: (VP:(or (and F1 ... Fn))
+    // premises: (VPn:(or (and F1 ... Fn) (not Fn))) Fn
+    // args: () 
+    case PfRule::AND_INTRO:{
+      bool success = true;
+      std::vector<Node> neg_Nodes;
+      neg_Nodes.push_back(d_nm->mkNode(kind::AND,children));
+      for(int i = 0; i < children.size(); i++){
+        neg_Nodes.push_back(d_nm->mkNode(kind::NOT,children[i]));
+      }
+
+      Node and_neg_Node = d_nm->mkNode(kind::OR,neg_Nodes);	
+      success = success && addVeriTStep(and_neg_Node,VeriTRule::AND_NEG,{},{},*cdp);
+
+      //TODO: It seems as if all this is unnecessary since the resolution rule in VeriT can do a 
+      //chain or resolution steps. Thus, this simplifies to a single resolution step and there is no need 
+      //to build up all these clauses.
```

Comment:
That's right. You could directly do a resolution in which the first premise is `and_neg_Node` and the others are the elements in `children`. One thing that we should aim for with the use of `VeriTRule::RESOLUTION` is to keep the correct order. This is one the first things that should be updated in the proofnomicon (as not being able to assume the proper order in resolution makes it NP-complete to check...).

Reply:
@HanielB What is the proper order? E.g. in the AND_INTRO case:
`(or (and F1 ... Fn) (not F1) ... (not Fn))`
There is no way to reorder this to:
`(or (not F1) ... (not Fn) (and F1 ... Fn))`
What is possible (not necessary here but ,maybe somewhere else) is to reorder the order of the children (which are the premises in veriT), so if you had
`(or (and F1 ... Fn) (not F2) (not F1) ... (not Fn))`
then  
`Children: (P2:F2 P1:F1 ... Pn:Fn)`

Reply:
This is a good question. :) The "order" I was worried about was when there are more than two clauses being resolved and you have C1 resolved w/ C2 resolved w/ C3 recorded in the proof as C2, C1, C3, say. The  "order" you exemplified above is for the literals in the clauses themselves, which is less serious (if clauses are viewed as sets, this does not add overhead, while in my example the issue is search the correct clause to resolve).

Currently the veriT proof format is agnostic to both "orders", but IIRC the Isabelle reconstruction assumes the correct order for the clauses being resolved, most of the time. So we should respect that. Does that make sense?